### ISSUE TITLE
Cheats not working for some games

### ISSUE THREAD

**Issue by Lacept** [DeadSkullzJr/NDS-Cheat-Databases#3]
Hi, I am running this with twilight menu ++ 10.3.0 (latest update as of this post), and whenever I try any of the cheats for pokemon platinum the game won't load. I tried the cheats with new super Mario bros and it worked perfectly.

**Comment by jmfindorff** [comment]
Yeah, I have the same issue with Platinum on my R4i Gold 3DS RTS cart

**Comment by johnnyboy0g** [comment]
How do you go about editing .dat files exactly? Mine currently just shows a bunch of nulls.

**Comment by jmfindorff** [comment]
That happens when you try to edit the file with a normal text file editor.
You need a special program to do it. R4cce seems to be the best program I
could find. It's an old program and I can't provide a link to it at the
moment because I had to use web.archive.org to access the now dead link.

**Comment by DeadSkullzJr** [comment]
Interesting issue, I personally have Platinum on my R4i Gold 3DS Plus (successor to the Gold RTS model) and the full cheat list loads fine for me. I recommend checking/verifying the checksum of the game to see if maybe the database just doesn't like whatever dump you are using.

**Comment by Anonymoussa** [comment]
I'm having a similar issue with Pokemon White 2. The game will not boot up with the "Wild Shiny Encounters" code enabled. I've tried multiple ROMs but haven't succeeded. I haven't messed around with most other codes but the few others I have tried work fine, so it's not an issue with the ROM or likewise. Commented on this one as I don't believe this is worth opening a new topic over.

**Comment by jwalker040** [comment]
hey guys i have the same problem mit Pokemon soul silver, when i chose any cheat the game don´t load, i tried many roms but the problem stays the same. for pokemon platin the cheats works fine! but there is no shiny cheat

**Comment by DeadSkullzJr** [comment]
> I'm having a similar issue with Pokemon White 2. The game will not boot up with the "Wild Shiny Encounters" code enabled. I've tried multiple ROMs but haven't succeeded. I haven't messed around with most other codes but the few others I have tried work fine, so it's not an issue with the ROM or likewise. Commented on this one as I don't believe this is worth opening a new topic over.



> hey guys i have the same problem mit Pokemon soul silver, when i chose any cheat the game don´t load, i tried many roms but the problem stays the same. for pokemon platin the cheats works fine! but there is no shiny cheat

Check your dumps for the game. Also depending one what you use, certain codes won't work due to compatibility issues via the method you used to use cheats on. TWiLightMenu to be exact doesn't have compatibility for one of the important code types, thus any codes using said type won't work. Certain flashcart kernels also have some issues with cheats as well.

**Comment by NotStarRift** [comment]
I'm having the same issue on my R4 Gold Pro cartridge. The cheats for Mario Kart DS are displaying incorrectly or won't "apply" to the rom when loading. WarioWare D.I.Y. cheats also don't seem to load at all.

**Comment by NotStarRift** [comment]
It might be that the subfolders are holding more cheats than they are designed to. I'll try splitting up the large folders and report back here.

**Comment by DeadSkullzJr** [comment]
> I'm having the same issue on my R4 Gold Pro cartridge. The cheats for Mario Kart DS are displaying incorrectly or won't "apply" to the rom when loading. WarioWare D.I.Y. cheats also don't seem to load at all.



> It might be that the subfolders are holding more cheats than they are designed to. I'll try splitting up the large folders and report back here.

It has everything to do with the flashcart kernel, if you had a flashcart running something like Wood R4 then you'd see for yourself that the databases display fine. The issue is the kernel used by most of the R4 flashcarts today are actually bugged, the cheat functions are broken with them, I suggest looking into YSMenu, it's an alternative kernel for time bomb flashcarts like the R4i SDHC Gold Pro.

**Comment by NotStarRift** [comment]
Thanks for the quick response. I'll definitely look into that!

**Comment by NotStarRift** [comment]
So I got YSMenu fully working on my R4 cartridge and flashed your cheat database but unfortunately some games still don't display the cheat option even though the correct region is there and everything!

**Comment by DeadSkullzJr** [comment]
> So I got YSMenu fully working on my R4 cartridge and flashed your cheat database but unfortunately some games still don't display the cheat option even though the correct region is there and everything!

Then that means the dumps that have that issue are likely bad dumps. That just means that the header information of the dump you have doesn't match the information in the databases. I don't support bad dumps unless I really have to, the only times I do is if a bad dump of a certain title is the only dump available for that title, or if a ROM hack uses a bad dump as a base. You'll have to get your hands on proper dumps, whether you dump the games yourself, or if you find them someplace else (obviously dumping your own games is more legal compared to the other option).
